src/main/java/com/mastercard/app/petstore/utils/OAuth2Utils.java

```diff
@@ -0,0 +1,161 @@
+package com.mastercard.app.petstore.utils;
+
+import com.mastercard.developer.encryption.EncryptionConfig;
+import com.mastercard.developer.interceptors.OkHttpJweInterceptor;
+import com.mastercard.developer.oauth2.interceptors.OkHttpOAuth2Interceptor;
+import com.mastercard.developer.utils.AuthenticationUtils;
+import okhttp3.Interceptor;
+import org.openapitools.client.ApiClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Profile;
+
+import java.security.PrivateKey;
+
+@Profile({"oauth2", "oauth2local"})
+@org.springframework.context.annotation.Configuration
+public class OAuth2Utils {
+
+    private final String signingKeyContainer;
+    private final String signingKeyAlias;
+    private final String signingKeyPassword;
+    private final String basePath;
+    private final String tokenUrl;
+    private final String audience ;
+    private final String keyId;
+    private final String clientId;
+    private final String scope;
+    private final String dpopKeyType;
+
+    @Autowired
+    public OAuth2Utils(
+            @Value("${mastercard.oauth2.pkcs12KeyFile}") String signingKeyContainer,
+            @Value("${mastercard.oauth2.keyAlias}") String signingKeyAlias,
+            @Value("${mastercard.oauth2.keyPassword}") String signingKeyPassword,
+            @Value("${mastercard.basePath}") String basePath,
+            @Value("${mastercard.oauth2.audience}") String audience,
+            @Value("${mastercard.oauth2.tokenUrl}") String tokentUrl,
+            @Value("${mastercard.oauth2.keyId}") String keyId,
+            @Value("${mastercard.oauth2.clientId}") String clientId,
+            @Value("${mastercard.oauth2.scope}") String scope,
+            @Value("${mastercard.oauth2.dpopKeyType}") String dpopKeyType) {
+
+        if (basePath.isEmpty()){
+            throw new IllegalArgumentException("basePath in application.properties is empty");
+        }
+        this.basePath = basePath;
+        if (signingKeyContainer.isEmpty()){
+            throw new IllegalArgumentException("pkcs12KeyFile in application-oauth2.properties is empty");
+        }
+        this.signingKeyContainer = signingKeyContainer;
+        if (clientId.isEmpty()){
+            throw new IllegalArgumentException("client id  in application-oauth2.properties is empty");
```

Comment:
Should be `client id in application-oauth2.properties is empty` -> has two spaces